### PR TITLE
Added mentions.members

### PR DESCRIPTION
**Pull request by Drahcirius**
## Description
I have added a convenience field to mentions that maps guild members to their respective mentions, provided they already exist in the cache.
I also changed the users code to be a bit neater, but the functionality is exactly the same.

### Case 1
Frequently when doing operations on mentions, we need to map them to the member get function.
```js
if (message.mentions.users.size) {
	const x = message.mentions.users.map(u => message.guild.members.get(u.id).displayName);
	message.reply(`Members: ${x}`);
}
```
Using the convenience property, we can shorten this to:
```js
if (message.mentions.members.size) {
	const x = message.mentions.members.map(m => m.displayName);
	message.reply(`Members: ${x}`);
}
```
### Case 2
For moderation purposes such as checking for kickable:
```js
message.guild.members.get(message.mentions.users.first().id).kickable;
```
Would become:
```js
message.mentions.members.first().kickable;
```

## Things to note

- If a member is not in the cache, the members list will not be populated with their entry. This can lead to strange behaviour such as where the `users.first()` and `members.first()` may not be the same entry.
- ~~In a DM, this field will be an empty collection.~~ ~~Amended to be undefined for non-guild channels.~~ Further amended to return empty collection again

### CODE REVIEW

**Comment by GusCaplan** [comment]
add a thing in message#patch for this

**Comment by bdistin** [comment]
Might it be better if you wrap the definition of mentions.members in an `if (data.type === '0')` so that mentions.members is only defined/looped in a guild text channel?

**Comment by Drahcirius** [comment]
I have separated the definition of mentions.members and the adding members into its own loop.

**Comment by GusCaplan** [comment]
in the interest of keeping state down, i would suggest making mentions.members a getter based off of the current mentions.users so that we don't have to patch both and can keep consistency up
